### DISCUSSION TITLE
mitmproxy with wireguard mode doesn't work

### DISCUSSION THREAD

**Discussion by Memexurer** [mitmproxy/mitmproxy#5800] [Q&A]
so, as the title says, i'm trying to capture some requests with mitmproxy 9.0.1 
i can't connect to anything while this error occurs, pls halp

https://user-images.githubusercontent.com/34003944/206799175-02b5240f-0e1e-4b7f-835a-fc0cd315ddfb.mp4

**Comment by mhils** [comment]
Are you running mitmproxy and WireGuard on the same device, i.e. are you trying to capture the traffic of the device mitmproxy is running on?

**Reply by Memexurer** [reply]
yup

**Reply by mhils** [reply]
That's currently not supported unfortunately as it runs into an infinite loop. We do have some plans to bring an even better transparent mode to Windows though, see https://github.com/mitmproxy/mitmproxy/discussions/5795. :)

**Reply by Memexurer** [reply]
btw, do you know why wireguard mode doesn't work with wiresock? (https://www.wiresock.net/) 
i tested wiresock with other wireguard servers and it worked, but it didn't work with the server from mitmproxy
![image](https://user-images.githubusercontent.com/34003944/206850831-04e5e43a-cf66-4e41-b084-2f4243fe6167.png)
here's how i configured it

**Reply by Memexurer** [reply]
nvm, i don't need it anymore. i just made an mitmproxy wireguard server on other machine.
